relay-server/src/actors/processor.rs

```diff
@@ -1352,6 +1363,14 @@ impl EnvelopeProcessorService {
         //  2. The DSN was moved and the envelope sent to the old project ID.
         envelope.meta_mut().set_project_id(project_id);
 
+        #[allow(unused_mut)]
+        let mut reservoir = ReservoirEvaluator::new(reservoir_counters);
+        #[cfg(feature = "processing")]
+        if let Some(redis_pool) = self.inner.redis_pool.as_ref() {
+            let org_id = managed_envelope.scoping().organization_id;
+            reservoir = reservoir.set_redis(org_id, redis_pool.clone());
```

Comment:
_nit_: Since we need this in a branch, what if we change the signature to `set_redis(&mut self, ...)` and no return value?